### PR TITLE
refactor(net): Simplified logic for determining shared capability versions

### PR DESCRIPTION
**Pull request by tcoratger**
## Description

Refactored the logic to determine shared capability versions, replacing complex conditionals with a more concise approach using `Option`'s `map_or` and `unwrap_or`. This enhances readability and maintainability while achieving the same functionality.

### CODE REVIEW

File: crates/net/eth-wire/src/capability.rs [hunk 1]
```diff
@@ -479,12 +479,8 @@ pub fn shared_capability_offsets(
...
             // If multiple versions are shared of the same (equal name) capability, the numerically
             // highest wins, others are ignored
-
-            let version = shared_capabilities.get(&peer_capability.name).map(|v| v.version);
-
-            // TODO(mattsse): simplify
-            if version.is_none() ||
-                (version.is_some() && peer_capability.version > version.expect("is some; qed"))
+            if peer_capability.version >
+                shared_capabilities.get(&peer_capability.name).map(|v| v.version).unwrap_or(0)
```
Comment (mattsse) [inline crates/net/eth-wire/src/capability.rs:483]: there's an edge case here when the shared version == 0.

I'd like to see a test for a cap with version 0 and I think we can fix this by moving the comparison into the map and use `map_or(true, )`

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm
